### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+## Top level pipeline jobs
+
+These live at https://ci.adoptopenjdk.net/job/build-scripts/. Currently these jobs are:
+
+* [openjdk8-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/)
+* [openjdk9-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk9-pipeline/)
+* [openjdk10-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk10-pipeline/)
+* [openjdk11-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk11-pipeline/)
+
+From he on in this documentation describes the 10 pipeline however it is the same for all Java versions.
```
Comment (karianna) [inline docs/build.md]: typo `here on` is what I thin you wanted

File: README.md [hunk 2]
```diff
@@ -277,3 +277,31 @@ to your local platform and then build (Adopt) OpenJDK and place the result in
 
 These scripts default to using AdoptOpenJDK as the OpenJDK source repo to build 
 from, but you can override this with the `-r` flag.
+
+
+# Build farm
+
+The documentation for files that define the build farm builds can be found [here](/docs/build.md).
```
Comment (karianna) [inline README.md]: Usually better for SEO to have the link say `Build Farm Definition` as opposed to `here`

File: docs/build.md (hunk 1 above)
Comment (karianna) [reply docs/build.md]: and say openjdk10 pipeline

File: docs/build.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+The high level sequence of each of these pipelines is shown in:
+
+![build sequence](images/sequence.svg "Build Sequence")
+
+This shows a build of `jdk10-mac-x64-hotspot` and `jdk10-linux-x64-hotspot`. The `jdk10-mac-x64-hotspot` build is expanded as
+an example of how each build pipeline works, however a similar pipeline will be generated for every `jdk10-<os>-<arch>-<impl>`
```
Comment (karianna) [inline docs/build.md]: wrapped whitespace

File: docs/build.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+The actions of the pipeline are defined by [openjdk10_pipeline.groovy](/pipelines/build/openjdk10_pipeline.groovy), which 
+calls [BuildBaseFile.groovy](/pipelines/build/BuildBaseFile.groovy).
+
+The actions of this script are:
+
+1. Read in build configuration that determines what configurations have been requested to be built.
+1. For each requested configuration generate a configuration that defines such things as which nodes to run the build on, optional configuration added to [openjdk10_pipeline.groovy](/pipelines/build/openjdk10_pipeline.groovy) etc.
+1. Generate a build job for each requested configuration, these live at [jdk10u](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk10u/) 
+1. Execute all build jobs in parallel
+1. Publish binaries to github
```
Comment (karianna) [inline docs/build.md]: GitHub

File: docs/build.md (hunk 1 above)
Comment (karianna) [reply docs/build.md]: Might be worth bolding that you're using 10 as an example here.

File: docs/build.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+
+#### Generated build jobs
+
+These are generated using the template at [createJobFromTemplate.groovy](/pipelines/build/createJobFromTemplate.groovy)
+This defines a pipeline job that calls [openjdk_build_pipeline.groovy](/pipelines/build/openjdk_build_pipeline.groovy), this script defines the work of what a Build Pipeline does.
```
Comment (karianna) [inline docs/build.md]: idle thought - if these pipelines are both build and test we should maybe drop the `build` wording in their file names

File: docs/build.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+
+#### Generated build jobs
+
+These are generated using the template at [createJobFromTemplate.groovy](/pipelines/build/createJobFromTemplate.groovy)
+This defines a pipeline job that calls [openjdk_build_pipeline.groovy](/pipelines/build/openjdk_build_pipeline.groovy), this script defines the work of what a Build Pipeline does. 
+
+## Build Pipelines
+
+Generated by [createJobFromTemplate.groovy](/pipelines/build/createJobFromTemplate.groovy), and its work defined in [openjdk_build_pipeline.groovy](/pipelines/build/openjdk_build_pipeline.groovy).
```
Comment (karianna) [inline docs/build.md]: just realised that the file naming std is different, some camel case others underscore separated.

File: docs/build.md (hunk 5 above)
Comment (karianna) [reply docs/build.md]: Also see below

File: docs/build.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+The actions of this are:
+
+1. Execute build
+1. Spawn requested test pipelines, these are defined in [openjdk10_pipeline.groovy](/pipelines/build/openjdk10_pipeline.groovy)
```
Comment (karianna) [inline docs/build.md]: so these are test pipelines only? Maybe then we should add `test` to their names.

File: docs/generateBuildMatrix.sh [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+echo "| Platform                  | Java 8 | Java 9 | Java 10 | Java 11 |"
+echo "| ------------------------- | ------ | ------ | ------- | ------- |"
+
+rm "/tmp/build.txt"
+for i in "8u" "9u" "10u" "11";
+do
+    curl -s "https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk${i}/" | egrep -o "job/jdk${i}u?-[^\/]+" >> "/tmp/build.txt"
+done
+
```
Comment (karianna) [inline docs/generateBuildMatrix.sh]: kill some blank lines

File: docs/generateBuildMatrix.sh [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+
+cat "/tmp/build.txt" | cut -d'/' -f2 | sed -r 's/jdk[0-9]+u?\-//g' | sort | uniq  | while read buildName;
```
Comment (karianna) [inline docs/generateBuildMatrix.sh]: extra whitespace after uniq

Comment (karianna) [reply docs/generateBuildMatrix.sh]: And this wins an award for piping glory :-D

Comment (karianna) [reply docs/generateBuildMatrix.sh]: Would be worth putting a sample row in as a commment above

File: docs/build.md (hunk 5 above)
Comment (johnoliver) [reply docs/build.md]: Its really a build that then triggers a test

File: docs/build.md (hunk 7 above)
Comment (johnoliver) [reply docs/build.md]: no those are top level

**Pull request merged**